### PR TITLE
Feature xs3 verification and build support

### PR DESCRIPTION
**Pull request by ed-xmos**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -3,57 +3,193 @@
...
+                    "spi_master_sync_multi_client",
+                    "spi_master_sync_multi_device",
+                    "spi_master_sync_rx_tx",
+                    "spi_slave_benchmark",
+                    "spi_slave_rx_tx"
+                  ]
+                  tests.each() {
+                    dir(it) {
+                      //Build all test under XS3 to check for build errors
+                      runXmake(".", "", "XCOREAI=1")
```
Comment (danielpieczko) [inline Jenkinsfile:81]: Where does this get run? I can't see if in http://srv-bri-jcim0:8080/blue/organizations/jenkins/XMOS%2Flib_spi/detail/feature%2Fxs3_support/38/pipeline

**Review by danielpieczko — COMMENTED** [review]
Just some questions on the Jenkinsfile changes.

File: Jenkinsfile [hunk 2]
```diff
@@ -3,57 +3,193 @@
...
       }
-    }
-    stage('Library checks') {
-      steps {
-        xcoreLibraryChecks("${REPO}")
+
+      environment {
+        REPO = 'lib_spi'
+        // VIEW = "${env.JOB_NAME.contains('PR-') ? REPO+'_'+env.CHANGE_TARGET : REPO+'_'+env.BRANCH_NAME}"
+        VIEW = "lib_spi_feature_xs3_support"
```
[the remark this answers is not in the document]
Comment (ed-xmos) [reply Jenkinsfile:23]: Yes exactly, but a good spot. I need to revert this.

File: Jenkinsfile [hunk 3]
```diff
@@ -3,57 +3,193 @@
...
+            xcoreLibraryChecks("${REPO}")
+          }
+        }
+        stage('Legacy tests') {
+          steps {
+            dir("${REPO}/legacy_tests") {
+              viewEnv() {
+                // Use Pipfile in legacy_tests, not lib_spi/Pipfile
+                // installPipfile(true)
+                // runPython("./runtests.py --junit-output=${REPO}_tests.xml")
```
[the remark this answers is not in the document]
Comment (ed-xmos) [reply Jenkinsfile:45]: Thanks! Forgot to uncomment. Will fix

File: Jenkinsfile (hunk 1 above)
Comment (ed-xmos) [reply Jenkinsfile:81]: These tests do not get run. However they are built, so at least the compiler checks them. This is mentioned here:

https://github.com/xmos/lib_spi/issues/48

Comment (danielpieczko) [reply Jenkinsfile:80]: Thank you for the explanation.

**Review by danielpieczko — APPROVED** [review]
Looks good.

**Pull request merged**